### PR TITLE
sql: don't startExec planHook subplans

### LINKED ISSUE #29024: importccl,distsql: EXPORT with ORDER BY fails with column-in-index error
From @a-robinson's QA of `EXPORT` in #25223, originally in a tpch 100 table, but reproduced here in isolation:
```
root=> create table foo (i int primary key, x int, y int, z int, index (y));
CREATE TABLE
root=> insert into foo values (1, 12, 13, 14), (2, 22, 23, 24), (3, 32, 33, 34);
INSERT 0 3
root=> EXPORT INTO CSV 'gs://cockroach-tmp/dt/export-1' from select * from foo order by y asc limit 1;
ERROR:  requested column x not in index
```

### PR DESCRIPTION
**Pull request by dt**
Sub-plans of hooked plans are never run locally, so they do not want to be started.
Sub-plans are only ever used as the basis for distsql planning which is run my the
hook functions directly.

Fixes #29024.

Release note: none.

### CODE REVIEW

**Comment by knz** [comment]
Removing myself from the review because I simply don't know how this works.

**Comment by dt** [comment]
TLDR: we used to have hook plans pass their subplans back along with the hook fn so that the planner could do something to them. Specifically, the thing we did to them was startExec them before we started the hook fn. We now no longer do anything to them. EXPORT is the only hook with subplans, and it seems happy with this.

**Comment by jordanlewis** [comment]
LGTM assuming you've done sufficient extra manual testing (?) to determine that this is fine. The code makes sense to me on principle and it appears correct.

**Review by jordanlewis — APPROVED** [review]

**Review by madelynnblue — COMMENTED** [review]
Export test LGTM but I'm not qualified to review the rest.
